### PR TITLE
MODsuits

### PR DESCRIPTION
**Pull request by JerryWester**
## About The Pull Request

MODsuits, pulled straight from [/tg/station](https://tgstation13.org/wiki/MODsuits). Ported from a mix of [/tg/station](https://github.com/tgstation/tgstation), [Skyrat](https://github.com/Skyrat-SS13/Skyrat-tg), and an unfinished [Voidcrew port](https://github.com/voidcrew/Voidcrew-LRP/pull/430)

## Why It's Good For The Game

Because MODsuits are cool and fun and, most importantly, entirely optional and don't replace hardsuits or voidsuits

## Changelog
:cl:
add: MODsuits, MODsuits, MODsuits. Construction materials, research nodes, modules, etc.
add: Broken MOD cores can be found in maintenance loot
tweak: Defibs fire a signal on a successful result
tweak: AI and pAI procs to enable MODsuit control
wip: Not all modules are yet implemented, and some of them are hackish workarounds (see kinesis module)
wip: Not all modules are thoroughly tested
/:cl:

### CODE REVIEW

**Comment by JerryWester** [comment]
Not all modules are in, but it should be stable enough for a review

File: code/game/objects/structures/crates_lockers/closets/secure/scientist.dm [hunk 1]
```diff
@@ -31,4 +31,6 @@
 	new /obj/item/circuitboard/machine/techfab/department/science(src)
 	new /obj/item/storage/photo_album/RD(src)
 	new /obj/item/clothing/suit/hooded/wintercoat/rd(src)
+	new /obj/item/analyzer/hilbertsanalyzer(src)
```
Comment (SandPoot) [inline code/game/objects/structures/crates_lockers/closets/secure/scientist.dm]: Don't think this belongs here

Comment (JerryWester) [reply code/game/objects/structures/crates_lockers/closets/secure/scientist.dm]: Gah, `cherry-pick` has failed me. Fixed

File: code/game/objects/items/devices/aicard.dm [hunk 2]
```diff
@@ -26,18 +26,16 @@
...
+/obj/item/aicard/pre_attack(atom/target, mob/living/user, params)
 	if(AI) //AI is on the card, implies user wants to upload it.
-		log_combat(user, AI, "carded", src)
 		target.transfer_ai(AI_TRANS_FROM_CARD, user, AI, src)
 	else //No AI on the card, therefore the user wants to download one.
 		target.transfer_ai(AI_TRANS_TO_CARD, user, null, src)
 		if(AI)
-			log_combat(user, AI, "carded", src)
-	update_icon() //Whatever happened, update the card's state (icon, name) to match.
+			log_combat(user, AI, "uploaded", src, "to [target].")
```
Comment (Linzolle) [inline code/game/objects/items/devices/aicard.dm]: the wording on this log seems backwards... also the removal of the uploading log, it seems like this message was supposed to go in the first case and should've remained unchanged here

File: code/game/objects/items/singularityhammer.dm [hunk 3]
```diff
@@ -53,7 +53,13 @@
 			var/atom/movable/A = X
 			if(A == wielder)
 				continue
-			if(A && !A.anchored && !ishuman(X))
+			if(isliving(A))
+				var/mob/living/vortexed_mob = A
+				if(vortexed_mob.mob_negates_gravity())
+					continue
+				else
+					vortexed_mob.Paralyze(2 SECONDS)
```
Comment (Linzolle) [inline code/game/objects/items/singularityhammer.dm]: weird seemingly unrelated singularity hammer buff? am i missing something here?

Comment (Linzolle) [reply code/game/objects/items/singularityhammer.dm]: i mean it's a buff that... doesn't really affect players but odd change being here

**Comment by shellspeed1** [comment]
One thing I would recommend so far, have all of the modules and components at robotics instead of spread to different lathes. It saves on running and gives people a reason to interact with robotics if they aren't silicons which is nice.

**Comment by JerryWester** [comment]
> One thing I would recommend so far, have all of the modules and components at robotics instead of spread to different lathes. It saves on running and gives people a reason to interact with robotics if they aren't silicons which is nice.

Included, though I did wish to avoid giving roboticists too much power by allowing them access to virtually every module

**Pull request merged**